libsolidity/analysis/TypeChecker.cpp

```diff
@@ -551,33 +551,18 @@ void TypeChecker::endVisit(InheritanceSpecifier const& _inheritance)
 
 	if (arguments)
 	{
-		bool v050 = m_scope->sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::V050);
-
 		if (parameterTypes.size() != arguments->size())
 		{
-			if (arguments->size() == 0 && !v050)
-				m_errorReporter.warning(
-					_inheritance.location(),
-					"Wrong argument count for constructor call: " +
-					toString(arguments->size()) +
-					" arguments given but expected " +
-					toString(parameterTypes.size()) +
-					"."
-				);
-			else
-			{
-				m_errorReporter.typeError(
-					_inheritance.location(),
-					"Wrong argument count for constructor call: " +
-					toString(arguments->size()) +
-					" arguments given but expected " +
-					toString(parameterTypes.size()) +
-					"."
-				);
-				return;
-			}
+			m_errorReporter.typeError(
+				_inheritance.location(),
+				"Wrong argument count for constructor call: " +
+				toString(arguments->size()) +
+				" arguments given but expected " +
```

Comment:
Sorry, forgot about that in the last review: We should also add something like `Remove parenthesis if you do not want to provide arguments here.`

Reply:
I think the suggestion makes sense here, even though I was a bit undecided about it been reported in this scenario: https://github.com/ethereum/solidity/pull/4454/commits/be29fd5b3c677b9492be8cb2f31ba8d27216e3cf#diff-c03f6d6e2ee8783cb11d4cf3034a52b6